package.json

```diff
@@ -17,25 +18,21 @@
     "test": "tests"
   },
   "scripts": {
-    "test": "gulp test",
-    "prepare": "gulp build",
-    "docs": "jsdoc2md build/{,**/}*.js > DOCUMENTATION.md"
+    "clean": "rimraf build",
+    "build": "npm run clean && tsc -p tsconfig.build.json",
+    "lint": "balena-lint --typescript lib tests",
+    "test": "npm run build && npm run lint && npm run test:fast",
+    "test:fast": "mocha",
+    "docs": "jsdoc2md build/{,**/}*.js > DOCUMENTATION.md",
+    "prepack": "npm run build"
   },
-  "author": "Juan Cruz Viotti <juan@balena.io>",
+  "author": "Balena.io. <hello@balena.io>",
   "license": "Apache-2.0",
-  "devDependencies": {
-    "coffeescript": "^1.12.7",
-    "gulp": "^4.0.2",
-    "gulp-coffee": "^2.3.5",
-    "gulp-coffeelint": "^0.6.0",
-    "gulp-mocha": "^4.3.1",
-    "gulp-util": "^3.0.8",
-    "jsdoc-to-markdown": "^5.0.3",
-    "mocha": "^3.5.3",
-    "mochainon": "^2.0.0",
-    "mock-fs": "^4.12.0"
-  },
   "dependencies": {
+    "@types/jsonstream": "^0.8.30",
+    "@types/rsync": "^0.4.30",
+    "@types/ssh2": "^0.5.49",
+    "@types/tar-fs": "^2.0.1",
```

Comment:
Hmm these should have been in the `devDependencies` section rather than `dependencies` (type packages are not needed in downstream production installs). Spotted it when looking at the implications over the `balena-cli` `npm-shrinkwrap.json` file in balena-io/balena-cli/pull/2400.

Reply:
@pdcastro Why are type packages not needed? This isn't my understanding, because the exported type files may depend on these imported types...

Reply:
Looking at all occurrences of `'jsonstream'`, `'rsync'`, `'ssh2'` and `'tar-fs'` in the `balena-sync` source code, I have not spotted examples of those types being used in the module's public interface, suggesting they are `devDependencies`. I may of course have missed something.

In some situations, for limited usage of types of dependency modules in a module's public interface, downstream packages can also extract type information by using Typescript constructs like `typeof` or `ReturnType<someFunction>`. A couple of examples:

* https://github.com/balena-io/balena-cli/blob/v12.55.6/lib/utils/compose_ts.ts#L1398
* https://github.com/balena-io/balena-cli/blob/v12.55.6/lib/commands/devices/supported.ts#L117

But I have not identified the need for that either.